frontend/src/app/modules/settings/user/user-profile-page/user-profile-page.component.ts

```diff
@@ -127,15 +127,11 @@ export class UserProfilePageComponent extends BaseComponent implements OnInit {
         this.userService
             .editUser(editedUser)
             .pipe(this.untilThis)
-            .subscribe(
-                (user) => {
-                    this.userService.updateUser(user);
```

Comment:
As I understand we replaced it by event and move update local storage to user service ?

Reply:
Yes, I believe the user information in `localStorage` should be updated upon any API call, which alters the user on the backend side. So, the `UserService` is an ideal place. 

`userChangedEvent$` is preferred for usage as it **pushes** the notification that the user in `localStorage` was changed. And thus, less chances that some component won't react to the fact the user was changed(exactly how it was with this header avatar issue). 

But the old way of subscribing to `UserService` methods and manipulating the response also works. 
I have not changed all the old calls to the favor of event, because it seems like out of scope of this issue already.